File: Scripts/laptop/ackermann-function/go/ackermann-function.go
package main

func Ackermann(m, n int) int {
	switch 0 {
	case m:
		return n + 1
	case n:
		return Ackermann(m - 1, 1)
	}
	return Ackermann(m - 1, Ackermann(m, n - 1))
}

func main() {
    r := int(0)
    for i := 0; i < 1000000; i++ {
        for m := 0; m <= 3; m++ {
            for n := 0; n <= 3; n++ {
                r = Ackermann(m, n)
            }
        }
    }
 _ = r
}

File: Scripts/laptop/ackermann-function/r/ackermann-function.r
ackermann <- function(m, n) {
  if ( m == 0 ) {
    n+1
  } else if ( n == 0 ) {
    ackermann(m-1, 1)
  } else {
    ackermann(m-1, ackermann(m, n-1))
  }
}

executeTask <- function(i) {
  for ( m in 0:3 ) {
    for ( n in 0:3 ) {
      ackermann(m, n)
    }
  }

	return(i+1)
}

r = 1

for ( i in 0:1000000 ) {
	r = executeTask(i + r)	
}
